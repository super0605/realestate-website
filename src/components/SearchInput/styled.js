import styled from "styled-components";

export const SearchWrap = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  & .ant-input-search {
    width: 300px;
    @media (max-width: 499px) {
      width: 250px;
    }
    @media (max-width: 380px) {
      width: 200px;
    }
    .ant-input-wrapper {
      .ant-input {
        border-top-left-radius: 20px;
        border-bottom-left-radius: 20px;
      }
      .ant-input-group-addon {
        border-top-right-radius: 20px;
        border-bottom-right-radius: 20px;
        .ant-input-search-button {
          border-top-right-radius: 20px;
          border-bottom-right-radius: 20px;
        }
      }
    }
  }
`;