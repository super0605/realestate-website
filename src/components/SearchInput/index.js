import React from "react";
import { Input } from "antd";
import { SearchWrap } from "./styled";

const { Search } = Input;

const SearchInput = () => {
  const onSearch = (value) => console.log(value);

  return (
    <SearchWrap>
      <Search
        placeholder="input search text"
        defaultValue="Barcelona"
        onSearch={onSearch}
        enterButton
      />
    </SearchWrap>
  );
};

export default SearchInput;
