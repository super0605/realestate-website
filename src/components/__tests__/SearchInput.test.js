import * as React from "react";
import { mount, shallow } from "enzyme";
import SearchInput from "../SearchInput";

describe("SearchInput Component", () => {
  let props;
  let mountedSearchInput;
  let wrapper;

  const renderWrapper = (bShallow) => {
    if (!mountedSearchInput) {
      mountedSearchInput = bShallow
        ? shallow(<SearchInput {...props} />)
        : mount(<SearchInput {...props} />);
    }
    return mountedSearchInput;
  };

  beforeEach(() => {
    wrapper = renderWrapper(true);
  });

  it("should render without problems", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
