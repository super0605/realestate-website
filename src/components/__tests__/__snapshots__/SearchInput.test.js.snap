// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Component should render without problems 1`] = `
<styled.div>
  <Search
    defaultValue="Barcelona"
    enterButton={true}
    onSearch={[Function]}
    placeholder="input search text"
  />
</styled.div>
`;
